src/box/lua/schema.lua

```diff
@@ -820,6 +821,24 @@ box.schema.space.create = function(name, options)
         end
     end
     local id = options.id
+    if type(options.temporary) == 'string' then
+        if options.temporary == 'meta' then
+            if not id then
+                local meta_id_min = box.schema.SPACE_ID_META_TEMPORARY_MIN
+                local meta_id_max = meta_id_min - 1
+                for _, t in _vspace:pairs(meta_id_min, box.index.GE) do
+                    local id = t.id
+                    if meta_id_max < id then
+                        meta_id_max = id
+                    end
+                end
+                id = meta_id_max + 1
```

Comment:
done

Reply:
BTW box.schema.create_space since 697123d0d5 no longer does what your example implies, and now creating a meta-temporary space will make all regular spaces after that to have ids `SPACE_ID_META_TEMPORARY_MIN + n`, which is contrary to the whole idea of having a special range of ids for meta-temporary spaces.

Do you think I should patch box_generate_space_id function to ignore meta-temporary spaces, or do I just scrap the special range thing and force the user to worry about conflicts?

Reply:
I think what they did is just a bug. That commit is just wrong. I will file a bug report.

Reply:
done